src/test/java/com/shippo/model/BatchTest.java

```diff
@@ -0,0 +1,132 @@
+package com.shippo.model;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.Test;
+
+import com.shippo.exception.APIConnectionException;
+import com.shippo.exception.APIException;
+import com.shippo.exception.AuthenticationException;
+import com.shippo.exception.InvalidRequestException;
+import com.shippo.model.Batch.BatchStatus;
+
+public class BatchTest extends ShippoTest {
+
+	final String id = "dd442663c24843068977704b1bd7d91d";
+
+	@Test
+	public void testAll() throws AuthenticationException, InvalidRequestException, APIConnectionException, APIException,
+			UnsupportedEncodingException, MalformedURLException {
+		Batch[] batches = Batch.all();
+		Batch batch = null;
+		for (Batch b : batches) {
+			if (b.getId().equals(id)) {
+				batch = b;
+				break;
+			}
+		}
+		assertEquals(batch.getStatus(), Batch.BatchStatus.VALID);
+	}
+
+	@Test
+	public void testValidGet()
+			throws AuthenticationException, InvalidRequestException, APIConnectionException, APIException {
+		Batch batch = Batch.get(id, 0, null);
+		assertEquals(batch.getStatus(), Batch.BatchStatus.VALID);
+	}
+
+	@Test
+	public void testValidGetParams()
+			throws AuthenticationException, InvalidRequestException, APIConnectionException, APIException {
+		Batch batch = Batch.get(id, 1, Batch.ShipmentStatus.CREATION_SUCCEEDED);
+		for (BatchShipment bs : batch.getBatchShipments().getShipments()) {
+			if (bs.getStatus() != BatchShipment.Status.VALID) {
+				fail(String.format("BatchShipment %s is not VALID. Its status is %s", bs.getId(), bs.getStatus()));
+			}
+		}
+	}
+
+	@Test(expected = InvalidRequestException.class)
+	public void testInvalidGet()
+			throws AuthenticationException, InvalidRequestException, APIConnectionException, APIException {
+		Batch.get("invalid_id", 0, null);
+	}
+
+	private BatchShipment addShipment()
+			throws AuthenticationException, InvalidRequestException, APIConnectionException, APIException {
+		Shipment shipment = (Shipment) ShipmentTest.getDefaultObject();
+		String[] ids = { (String) shipment.getObjectId() };
+		Batch batch = Batch.addShipments(id, ids);
+		for (BatchShipment bs : batch.getBatchShipments().getShipments()) {
+			if (bs.getShipment().equals(ids[0])) {
+				return bs;
+			}
+		}
+		return null;
+	}
+
+	@Test
+	public void testAddShipments()
+			throws AuthenticationException, InvalidRequestException, APIConnectionException, APIException {
+		assertNotNull(addShipment());
```

Comment:
I think the extra test would ensure no side effects occurred, for example, adding the shipment twice or removing another shipment.